internal/io/event.go

```diff
@@ -17,14 +16,16 @@ import (
 // a reasonable limit appropriate for handling CBOR payload in edgex-go.  More details could be found at
 // https://github.com/edgexfoundry/edgex-go/issues/2439
 
-func ReadAddEventRequestInBytes(w http.ResponseWriter, reader io.ReadCloser, size int64) ([]byte, errors.EdgeX) {
-	// use MaxBytesReader for limiting the size of incoming request bodies to avoid unexpected memory exhaustion
-	bytes, err := io.ReadAll(http.MaxBytesReader(w, reader, size))
+func CheckPayloadSize(reader io.Reader, size int64) ([]byte, errors.EdgeX) {
+	// Treat 0 as unlimit size
+	payload, err := io.ReadAll(reader)
```

Comment:
move io.ReadAll to http controller because message bus doesn't need this operation

Reply:
updated